rook/scripts/autoconfig.sh

```diff
@@ -0,0 +1,141 @@
+#!/usr/bin/env bash
+
+# fn <address> <subnet>
+addressInSubnet() {
+  python3 -c "import ipaddress; exit(0) if ipaddress.ip_address('${1}') in ipaddress.ip_network('${2}') else exit(1)"
+}
+
+# fn <key> <addresses>
+updatePeers() {
+  peers="$(yq4 ".commons * .clusters.${cluster} | .networkPolicies.${1} | .[].cidr" "${CK8S_CONFIG_PATH}/rook/values.yaml")"
+
+  local -a update
+
+  for address in ${2}; do
+    for peer in ${peers}; do
+      if addressInSubnet "${address}" "${peer}"; then
+        update+=("${peer}")
+        continue 2
+      fi
+    done
+
+    update+=("${address}/32")
+  done
+
+  old="$(yq4 -oj -I0 '[split(" ") | sort | .[] | {"cidr": .}]' <<< "$peers")"
+  new="$(yq4 -oj -I0 '[split(" ") | sort | unique | .[] | {"cidr": .}]' <<< "${update[*]}")"
+
+  if ! diff -u3 --color --label "current .commons * .clusters.${cluster} | .${1}" <(yq4 -P <<< "$old") --label "update .cluster.${cluster}.${1}" <(yq4 -P <<< "$new"); then
+    echo -n "apply update? [Y/n]: "
+    read -r reply
+    if [[ "${reply}" =~ ^(Y|y|)$ ]]; then
+      echo apply
+    fi
+    yq4 -i ".clusters.${cluster}.networkPolicies.${1} = ${new}" "${CK8S_CONFIG_PATH}/rook/values.yaml"
+  else
+    echo "- ${cluster}/networkPolicies/${1}: up to date"
+  fi
+}
+
+if [[ -z "${CK8S_CONFIG_PATH:-}" ]]; then
+  echo "err: missing CK8S_CONFIG_PATH" >&2
+  exit
+elif [[ -z "${1:-}" ]]; then
+  echo "err: missing cluster name" >&2
+  exit
+elif [[ "$(yq4 ".clusters | keys | [\"${1}\"] - . | length" "${CK8S_CONFIG_PATH}/rook/values.yaml")" == "1" ]]; then
+  echo "err: invalid cluster name" >&2
+  exit
+fi
+
+cluster="${1}"
+
+dnsAddresses="$(kubectl -n kube-system get svc kube-dns -ojsonpath='{.spec.clusterIPs[0]}')"
```

Comment:
could this also be coredns svc? 
for some reason on my dev cluster I do not have kube-dns svc

Reply:
It seems like the service should still be there (and we currently depend on it via node-local-dns if I'm correct) as the condition on the deletion in kubespray is if kubespray *doesn't* manage the *coredns* service. Confusing all around.

Reply:
On the new env I created with kubespray 22 the service `kube-dns` is not present, only `coredns`

Reply:
Now it takes `kube-dns` if it exists else `coredns`.